### PR TITLE
HTTP API base structure

### PR DESCRIPTION
**Pull request by yemel**

### CODE REVIEW

**Comment by maraoz** [comment]
LGTM

File: api/routes/v1.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+function initRouter(backend) {
+  var router = express.Router();
+
+  function mockResponse(req, res, next) {
+    res.send('This is a mocked response. Backed service is: ' + backend.status);
+  }
+
+  router.get('/blocks', mockResponse);
+  router.get('/block/:blockHash', mockResponse);
```
Comment (martindale) [inline api/routes/v1.js]: This should be plural, `/blocks/:blockHash`

File: api/routes/v1.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  router.get('/block-index/:height', mockResponse);
```
Comment (martindale) [inline api/routes/v1.js]: We should also consider adding this as metadata on the `/blocks` endpoint.

File: api/routes/v1.js (hunk 1 above)
Comment (maraoz) [inline api/routes/v1.js]: I was planning to heavily refactor the route handling, so I wouldn't mind merging this like this.
(my plan was to add middleware that handles each level of the path)

Comment (martindale) [inline api/routes/v1.js]: [Express 4.0's Router](http://expressjs.com/api.html#router) might do what you're looking for.  Totally okay with merging so long as it gets fixed before shipping!

Comment (maraoz) [inline api/routes/v1.js]: Ah, thanks for pointing that out. I'll check express 4.0 then :)

**Pull request merged**